flaml/automl.py

```diff
@@ -1488,6 +1623,25 @@ def _search(self):
                 logger.info(f'ensemble: {stacker}')
                 self._trained_estimator = stacker
                 self._trained_estimator.model = stacker
+            elif self._retrain_final:
+                # reset time budget for retraining
+                self._state.time_from_start -= self._state.time_budget
+                if (self._state.time_budget - self._state.time_from_start

```

Comment:
why still need to check the budget?

Reply:
to prevent too small budget used for retraining.